test/jdk/tools/jar/ValidatorTest.java

```diff
@@ -0,0 +1,335 @@
+                new EntryNameTestCase("..\\Hello.txt", false),
+                new EntryNameTestCase("C:\\Hello.txt", false),
+                new EntryNameTestCase("D:/Hello.txt", false),
+                new EntryNameTestCase("foo\\bar.txt", false),
+                new EntryNameTestCase("foo/bar.txt", true),
+                new EntryNameTestCase("foo/../bar.txt", false),
+                new EntryNameTestCase("foo/./bar.txt", true),
+                new EntryNameTestCase("..", false),
+                new EntryNameTestCase(".", false),
+                new EntryNameTestCase("/home/foo.txt", false),
+                new EntryNameTestCase("./home/foo.txt", true),
+                new EntryNameTestCase("../home/foo.txt", false),
+                new EntryNameTestCase("foo/bar/..", false),
+                new EntryNameTestCase("foo/bar/.", true),
+                new EntryNameTestCase("/foo/bar/../../myhome/bin", false),
+                new EntryNameTestCase("/foo/bar/././myhome/bin", false),
+                new EntryNameTestCase("myHome/..valid", true),
+                new EntryNameTestCase("myHome/.valid", true),
+                new EntryNameTestCase("..valid", true),
+                new EntryNameTestCase(".valid", true)
+        );
+    }
+
+    private List<String> createInvalidEntryJar(Path path) throws IOException {
+        System.out.printf("%n%n*****Creating Jar with the invalid entry names*****%n%n");
+        var out = new ByteArrayOutputStream(1024);
+        List<String> invalidEntryNames;
+        try (var zos = new ZipOutputStream(out)) {
+            invalidEntryNames = zipEntryPaths()
+                .filter(testCase -> {
+                        try {
+                            zos.putNextEntry(new ZipEntry(testCase.entryName()));
+                            var content = "Content of " + testCase.entryName();
+                            zos.write(content.getBytes(StandardCharsets.UTF_8));
+                            return !testCase.isValid();
+                        } catch (IOException ioe) {
+                            throw new UncheckedIOException(ioe);
+                        }
+                    })
+                .map(EntryNameTestCase::entryName).toList();
+            zos.flush();
+        } catch (UncheckedIOException uioe) {
+            throw uioe.getCause();
+        }
+        Files.write(path, out.toByteArray());
+        return invalidEntryNames;
+    }
+
+
+    @Test
+    public void testValidate() throws IOException {
+        var zip = Path.of("MultipleManifestTest.jar");
+        writeManifestAsFirstSecondAndFourthEntry(zip, true, true);
+        try {
+            jar("--validate --file " + zip.toString());
+        } catch (IOException e) {
+            var err = e.getMessage();
+            System.out.println(err);
+            Assertions.assertTrue(err.contains("Warning: 2 copies of META-INF/MANIFEST.MF is detected in local file header"));
```

Comment:
Given you are importing Assertions, you should be able to just use `assertTrue`

Reply:
I didn't use static import. Since that seems to be preferred, I changed to do static imports.